pages/auth/register.tsx

```diff
@@ -32,7 +32,10 @@ function Register() {
   }
 
   useEffect(() => {
-    if (user) router.push("/home")
+    if (user && !user?.verificationStatus) {
+      router.push("/auth/confirm-phone")
+      return
+    }
```

Comment:
the verification status is in this type
`export type Verification = "NotVerified" | "Verified"`

when creating the user, it saves "NotVerified" to the DB directly. So what you need to check here is whether the `user.verificationStatus === "NotVerified"` instead of whether the value is falsy